File: CodeForces/1849B.cpp
#include<bits/stdc++.h>

using namespace std;

int mod(int x,int k){
        if(x%k==0)
            return k;
        else return x%k;
}

bool comp(pair<int,int> v1, pair<int,int> v2){
    return  (v1.first > v2.first);
}

int main(){
    int testno,n,k,x;
    cin>>testno;
    
    for(int tests=0;tests<testno;tests++){
        cin>>n>>k;
        vector<pair<int,int>> arr(n);
        for(int i=0;i<n;i++){
            cin>>x;
            arr[i] = {mod(x,k),i+1};
        }
        sort(arr.begin(),arr.end(),comp<int>());
        for(int i=0;i<n;i++){
            cout<<arr[i].second<<" ";
        }
    }
        
}

File: CodeForces/1849B.py
if __name__ == "__main__":
    for testno in range(int(input())):
        inp = input()
        arr = inp.split()
        n = int(arr[0])
        k = int(arr[1]) 
        arr2 = [int(d) for d in str(input()).split()]
        arr3=[]
        def mod(l):
            if l[1]%k == 0:
                return k
            else:
                return l[1]%k
        for i, a in enumerate(arr2):
            arr3.append([i,a])
        arr3.sort(key=mod,reverse=True)
        for _ in arr3:
            print(_[0]+1,end=" ")
        print("")